atoma-confidential/src/key_management.rs

```diff
@@ -154,10 +178,65 @@ impl X25519KeyPairManager {
         let shared_secret = self.compute_shared_secret(&public_key);
         Ok(encrypt_plaintext(plaintext, shared_secret, salt)?)
     }
+
+    /// Returns the file path where the private key should be stored.
+    ///
+    /// This method constructs a path by:
+    /// 1. Starting from the current working directory
+    /// 2. Adding a "keys" subdirectory
+    /// 3. Adding the "dh_privkey" file name
+    ///
+    /// # Returns
+    /// * `PathBuf` - Path to the private key file: `./keys/dh_privkey`
+    ///
+    /// # Note
+    /// Currently uses a hardcoded path relative to the current directory.
+    /// This is primarily intended for development/testing purposes.
+    /// Production environments should use a more secure and configurable location.
+    fn get_key_file_path() -> std::path::PathBuf {
+        // Use a more appropriate path, possibly from config
+        std::env::current_dir()
+            .unwrap_or_default()
+            .join(KEY_FILE_DIR)
+            .join(KEY_FILE_NAME)
+    }
+
+    /// Writes the current private key to a file at the root directory.
+    ///
+    /// # Warning
+    /// This function is intended for development/testing purposes only.
+    /// Writing private keys to disk in production is a security risk.
+    ///
+    /// # Returns
+    /// * `Ok(())` if the write was successful
+    /// * `Err(KeyManagementError)` if the write failed
+    pub fn write_private_key_to_file(&self) -> Result<()> {
```

Comment:
hide behind a feature flag, so when we are going to deploy real code we will not use it by accident
maybe a feature flag like "dont_use_in_production"

Reply:
I think we need to always store the private key